src/iop/crop.c

```diff
@@ -1111,6 +1111,61 @@ static gchar *_aspect_format(gchar *original,
     return g_strdup_printf("%s  %4.2f", original, (float)adim / (float)bdim);
 }
 
+static void _crop_handle_flip(dt_iop_module_t *self, const dt_image_orientation_t mode)
+{
+  dt_iop_crop_params_t *p = self ? self->params : NULL;
+  if(!p) return;
+
+  dt_iop_crop_gui_data_t *g = self->gui_data;
+
+  float left, oleft, right, oright, top, otop, bottom, obottom;
+
+  ++darktable.gui->reset; // required here ?
+
+  left = oleft      = p->cx;
+  top = otop        = p->cy;
+  right = oright    = 1.0f - p->cw;
+  bottom = obottom  = 1.0f - p->ch;
+
+  dt_print(DT_DEBUG_ALWAYS, "[crop_handle_flip] origin: %1.3f %1.3f %1.3f %1.3f mode=%s%s%s",
+      oleft, oright, otop, obottom,
+          mode == ORIENTATION_ROTATE_CW_90_DEG  ? "ROTATE_CW_90_DEG"
+        : mode == ORIENTATION_ROTATE_CCW_90_DEG ? "ROTATE_CCW_90_DEG"
+        : mode == ORIENTATION_FLIP_HORIZONTALLY ? "FLIP_HORIZONTALLY"
+        : mode == ORIENTATION_FLIP_VERTICALLY   ? "FLIP_VERTICALLY"
+        : "UNKNOWN",
+      self->enabled ? "" : ", crop disabled",
+      g ? "" : ", without UI changes");
+
+  if(mode == ORIENTATION_FLIP_HORIZONTALLY)       {left = oright; right = oleft;}
+  else if(mode == ORIENTATION_FLIP_VERTICALLY)    {top = obottom; bottom = otop;}
+  else if(mode == ORIENTATION_ROTATE_CW_90_DEG)   {top = oleft;   right = otop;    bottom = oright; left = obottom;}
+  else if(mode == ORIENTATION_ROTATE_CCW_90_DEG)  {top = oright;  right = obottom; bottom = oleft;  left = otop;}
+
+  if(g)
+  {
+    dt_bauhaus_slider_set(g->cx, left);
```

Comment:
Something funny seems to be going on with the soft bounds. Often, they get set at the current value (such that the slider is at the max and then jumps when it is scrolled or whatever after an orientation change). Rather than figuring out what is going on, why not just adjust params and then update the gui from that?
```
static void _crop_handle_flip(dt_iop_module_t *self, const dt_image_orientation_t mode)
{
  dt_iop_crop_params_t *p = self ? self->params : NULL;
  if(!p) return;

  float tmp = p->cx;
  if(mode == ORIENTATION_FLIP_HORIZONTALLY)      {             p->cx = 1.f - p->cw; p->cw = 1.f - tmp;}
  else if(mode == ORIENTATION_FLIP_VERTICALLY)   {tmp = p->cy; p->cy = 1.f - p->ch; p->ch = 1.f - tmp;}
  else if(mode == ORIENTATION_ROTATE_CW_90_DEG)  {p->cx = 1.f-p->ch; p->ch = p->cw; p->cw = 1.f-p->cy; p->cy = tmp;}
  else if(mode == ORIENTATION_ROTATE_CCW_90_DEG) {p->cx = p->cy; p->cy = 1.f-p->cw; p->cw = p->ch; p->ch = 1.f-tmp;}
  dt_iop_gui_update(self);

  dt_dev_add_new_history_item(darktable.develop, self, self->enabled);
}
```

This still sometimes leads to incorrect soft limits, which was sometimes happening before anyway. I now realise that at the bottom of `gui_update` there needs to be a call to
`gui_changed(self, NULL, NULL);`

Reply:
Yes, much better.

Also avoids "fish-duplication". (When i observed these problems i tried to replicate current behaviour). Currently i am not sure if that is a bauhaus thing or if we set the limits here in crop in a wrong way.